### ISSUE TITLE
get_logs() shows incorrect metrics.

### ISSUE THREAD

**Issue by Practcdi** [pycaret/pycaret#852]
### Pycaret version :  2.2.1

GPU : False

Steps Included :
1) I have created a DT regressor model
2) After that I perform tuning on it and save the model eval plot

___
**Problem:**
When I checked model eval score using `pull()` its shows R2 = 0.9624 and in Prediction Error Plot its is exactly same R2 = 0.9624.

But when I checked `get_logs()` all 3 values were different!
you can see R2=0.9100 in `get_logs()` for tags.Source = tune_model
___

So why metrics values are so different in both cases for the same experiment?
Anyone have idea ?

Thanks ,

![image](https://user-images.githubusercontent.com/58843833/99374858-0c131400-28e9-11eb-81f6-59751f7a12c6.png)

**Comment by Yard1** [comment]
IIIRC the metrics in the logs are based on test set results, while the ones displayed are from cross validation. Can you use `predict_model` on the tuned model and see what you get?

**Comment by Practcdi** [comment]
**How can I check metrics for train data and its model eval plots?**
___

I want to perform a comparison between:
1)  **Metrics score** and model eval plot for **Train data** 
2)  **Metrics score** and model eval plot for **Test data**

**Comment by pycaret** [comment]
@Practcdi 

- `MLFlow` logs what is displayed in the Notebook, so for example when you use the `create_model` function the score grid displayed is cross-validated results across the N folds. When you use the `predict_model` function without explicitly passing the `data` parameter, it logs the score on the hold-out set i.e. your test set. 

- `plot_model` function shows the plot using the trained model on the test data. If for some reason you want to see plots on train data, you can pass `use_train_data = True` inside the `plot_model` function.

- To check the scores on test data simply pass the trained model inside the `predict_model` function without explicitly passing the `data` parameter inside the `predict_model`. It returns the labeled data as Pandas.DataFrame and displays the score grid as well which is calculated on test data.

- To check the metrics on the complete train data using the fitted model you can do something like this:

```
# using boston dataset as example
from pycaret.datasets import get_data
data = get_data('boston')

# init setup
from pycaret.regression import *
s = setup(data, target = 'medv')

# train linear regression
lr = create_model('lr')

# pull untransformed train row index from the data
train = data.iloc[get_config('X_train').index,]

# generate predictions using trained model
predictions = predict_model(lr, data=train)

# check metrics using pycaret.utils module 
from pycaret.utils import check_metric
check_metric(train['medv'], predictions['Label'],metric='MAE')

# you can replace the last step with sklearn.metrics if you would like. 
```

Hope this helps. Please close the issue if this answers your question.
